File: solutions/javascript/find_the_parity_outlier.js
function findOutlier(integers) {
    //your code here
    const results = integers.map((number) => {
        if (number % 2 == 0) {
            return { number: number, is_even: true, is_odd: false };
        } else {
            return { number: number, is_even: false, is_odd: true };
        }
    });

    const odds = results.filter((result) => {
        return result.is_even === false;
    });

    const evens = results.filter((result) => {
        return result.is_odd === false;
    });

    if (odds.length == 1) {
        return odds[0].number;
    } else {
        return evens[0].number;
    }
}

File: solutions/elixir/find_the_parity_outlier.exs
defmodule Kata do
  def find_outlier(integers) do
    require Integer

    numbers_assessment =
      integers
      |> Enum.map(fn number -> %{"number" => number, "is_even" => Integer.is_even(number)} end)

    frequencies =
      numbers_assessment
      |> Enum.frequencies_by(fn %{"is_even" => even} -> even end)

    if frequencies.true == 1 do
      [result] =
        numbers_assessment
        |> Enum.filter(fn %{"is_even" => even} -> even == true end)

      result["number"]
    else
      [result] =
        numbers_assessment
        |> Enum.filter(fn %{"is_even" => even} -> even == false end)

      result["number"]
    end
  end
end
